rotkehlchen/db/snapshots.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+from zipfile import ZipFile
+
+from rotkehlchen.accounting.export.csv import CSVWriteError
+from rotkehlchen.accounting.structures.balance import BalanceType
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.utils import DBAssetBalance, LocationData
+from rotkehlchen.types import Location, Timestamp
+
+
+class Snapshot:
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db_handler = db_handler
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir_path = Path(self.tmp_dir.name)
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
```

Comment:
Please be adding a docstring to most functions with a quick explanation of what it does (one line is fine) and then just as we do in other functions, mention the possible exceptions it may raise.

Mentioning exceptions in the docstring is a habit we have started doing which helps us understand what errors need to be handled where.

Do it for most functions in this file.